### PR TITLE
Add proposal link as rewards table custom property

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://app.charmverse.io/charmverse/if-a-milestone-is-created-via-proposal-put-the-proposal-title-as-a-custom-property-5483940115308825)

### CODE REVIEW

**Review by mattcasey — APPROVED** [review]

File: components/rewards/components/RewardProperties/RewardPropertiesForm.tsx [hunk 1]
```diff
@@ -53,7 +58,10 @@ export function RewardPropertiesForm({
 
     onChange(updates);
   }
-
+  const sourceProposalPage = (values as RewardWithUsers).sourceProposalPage;
+  const proposalLinkValue = sourceProposalPage
+    ? [getAbsolutePath(`/${sourceProposalPage.id}`, space?.domain), sourceProposalPage.title]
```
Comment (mattcasey) [inline components/rewards/components/RewardProperties/RewardPropertiesForm.tsx:63]: one thing, if we're using our Link component, then u dont need to get the absolute path

File: components/rewards/components/RewardProperties/RewardPropertiesForm.tsx [hunk 2]
```diff
@@ -101,7 +109,7 @@ export function RewardPropertiesForm({
             reward={values as unknown as BoardReward}
             onChange={(properties: RewardPropertiesField) => {
               applyUpdates({
-                fields: { properties: properties ? { ...properties } : {} } as Prisma.JsonValue
+                fields: { properties: { ...properties, [REWARD_PROPOSAL_LINK]: proposalLinkValue } } as Prisma.JsonValue
```
Comment (mattcasey) [inline components/rewards/components/RewardProperties/RewardPropertiesForm.tsx:112]: why do we need to include this when applying changes to properties?

File: components/rewards/components/RewardProperties/CustomPropertiesAdapter.tsx [hunk 3]
```diff
@@ -48,7 +50,17 @@ export function CustomPropertiesAdapter({ reward, onChange, readOnly }: Props) {
...
           // extract non-custom properties
-          cardProperties: board.fields.cardProperties.filter((p) => !p.id.startsWith('__'))
+          cardProperties: [
+            ...board.fields.cardProperties.filter((p) => !p.id.startsWith('__')),
+            // Add proposal link as a custom property
+            {
+              readOnly: true,
+              options: [],
+              id: REWARD_PROPOSAL_LINK,
+              name: 'Proposal',
```
Comment (mattcasey) [inline components/rewards/components/RewardProperties/CustomPropertiesAdapter.tsx:60]: Only change request: this needs to be translated based on the spaces feature name

File: components/rewards/components/RewardProperties/RewardPropertiesForm.tsx (hunk 2 above)
Comment (Devorein) [reply components/rewards/components/RewardProperties/RewardPropertiesForm.tsx:112]: Ah crap good point. I forgot to remove it after adding it later. I thought this was the way to apply the fields. Thnx for catching will update the code

File: components/rewards/components/RewardProperties/CustomPropertiesAdapter.tsx (hunk 3 above)
Comment (Devorein) [reply components/rewards/components/RewardProperties/CustomPropertiesAdapter.tsx:60]: Good point. Updated.

**Pull request merged**